egs/jnas/asr1/local/jnas_train_prep.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+# Copyright 2017 Nagoya University (Someki Masao)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+# Prepare JNAS dataset
+
+. ./path.sh
+
+if [ $# != 2 ]; then
+    echo "Usage: $0 <data-directory> <speaker_text>";
+    exit 1;
+fi
+
+DATA=$1  # database root directory
+speaker_path=$2
+
+echo "=== Starting initial JNAS training_data preparation ..."
+
+wavdir=${DATA}/WAVES_HS
+trans=Transcription  # transcription dir name
+type=KANJI  # transcription type
+ifNP=NP
+locdata=${PWD}/data
+loctmp=$locdata/train/tmp
+
+rm -rf $loctmp >/dev/null 2>&1
+mkdir -p ${locdata}/train/tmp
+
+# extract speakers names
+
+logdir=exp/train_prep
+mkdir -p $logdir
+echo -n > $logdir/make_trans.log
+
+echo "--- Preparing train/wav.scp, train/trans.txt and train/utt2spk ..."
+
+for spkname in $(cat ${speaker_path}/train_speakers.txt); do
```

Comment:
spaker_path should be the path of speaker list, not directory.
```suggestion
for spkname in $(cat ${speaker_path}); do
```